src/common/plugins/plugin-readiness-middleware.js

```diff
@@ -0,0 +1,54 @@
+//@flow
+
+import {BaseMiddleware, getLogger} from '@playkit-js/playkit-js';
+import {BasePlugin} from './base-plugin';
+class PluginReadinessMiddleware extends BaseMiddleware {
+  _plugins: Array<BasePlugin>;
+  id: string = 'PluginReadinessMiddleware';
+  static _logger = getLogger('PluginReadinessMiddleware');
+
+  constructor(plugins: Array<BasePlugin>) {
+    super();
+    this._plugins = plugins;
+    PluginReadinessMiddleware._logger.debug('plugins readiness', this._plugins);
+  }
+
+  /**
+   * Load middleware handler.
+   * @param {Function} next - The load handler in the middleware chain.
+   * @returns {void}
+   * @memberof PluginReadinessMiddleware
+   */
+  load(next: Function): void {
+    this.checkNextSettle(0, next);
+  }
+  checkNextSettle(index, next: Function) {
+    if (index < this._plugins.length) {
+      this.checkSettle(index, next);
+    } else {
+      this.callNext(next);
+    }
+  }
+  checkSettle(index: number, next: Function) {
```

Comment:
is this an API? if not add _ to signal as private.

Reply:
done